boxsdk/client/client.py

```diff
@@ -539,6 +539,251 @@ def create_group(self, name):
+            custom_notification_recipients=None,
+    ):
+        """
+        Create a retention policy for the given enterprise.
+
+        :param policy_name:
+            The name of the retention policy.
+        :type policy_name:
+            `unicode`
+        :param retention_length:
+            The amount of time in days to apply the retention policy to the selected content.
+            Do not specify for `indefinite` policies, only for `finite` policies.
+        :type retention_length:
+            `int` or float('inf')
+        :param disposition_action:
+            For `finite` policy can be set to `permanently delete` or `remove retention`.
+            For `indefinite` policy this must be set to `remove_retention`
+        :type disposition_action:
+            `unicode`
+        :param can_owner_extend_retention:
+            The owner of a file will be allowed to extend the retention if set to true.
+        :type can_owner_extend_retention:
+            `boolean` or None
+        :param are_owners_notified:
+            The owner or co-owner will get notified when a file is nearing expiration.
+        :type are_owners_notified:
+            `boolean` or None
+        :param custom_notification_recipients:
+            A custom list of user mini objects that should be notified when a file is nearing expiration.
+        :type custom_notification_recipients:
+            `list` or `User` objects
+        :return:
+            The newly created Retention Policy
+        :rtype:
+            :class:`RetentionPolicy`
+        """
+        url = self.get_url('retention_policies')
+        retention_attributes = {'policy_name': policy_name}
+        if retention_length == float('inf') or retention_length == -1:
+            retention_attributes['policy_type'] = 'indefinite'
+            retention_attributes['disposition_action'] = 'remove_retention'
+        else:
+            retention_attributes['policy_type'] = 'finite'
+            retention_attributes['retention_length'] = retention_length
+            retention_attributes['disposition_action'] = disposition_action
+
+        if can_owner_extend_retention is not None:
+            retention_attributes['can_owner_extend_retention'] = can_owner_extend_retention
+        if are_owners_notified is not None:
+            retention_attributes['are_owners_notified'] = are_owners_notified
+        if custom_notification_recipients is not None:
+            retention_attributes['custom_notification_recipients'] = custom_notification_recipients
+        box_response = self._session.post(url, data=json.dumps(retention_attributes))
+        response = box_response.json()
+        return self.translator.translate(response['type'])(session=self._session, object_id=response['id'], response_object=response)
+
+    def get_retention_policies(
+            self,
+            policy_name=None,
+            policy_type=None,
+            created_by_user_id=None,
```

Comment:
This should take a `User` object instead of an ID, for consistency with the rest of the SDK